File: Backend /BeerJournal.Model/Entities/TastingEntry.cs
namespace BeerJournal.Model.Entities;

public class TastingEntry
{
    public int EntryId { get; set; }
    public int UserId { get; set; }

    // Foreign key to the beers table — replaces the four old fields
    // (BeerName, Brand, AlcoholPct, Category)
    public int BeerId { get; set; }

    // Tasting info — these stay on the entry
    public int? Rating { get; set; }
    public string? Location { get; set; }
    public DateOnly? TestingDate { get; set; }
    public decimal? Price { get; set; }
    public string? Notes { get; set; }

    // Map coordinates
    public double? Latitude { get; set; }
    public double? Longitude { get; set; }

    // Nested beer object — populated when reading entries (via JOIN)
    // The frontend uses this to display beer name/brand/etc on each card
    // Nullable because on POST/PUT the client only sends BeerId, not the full beer
    public Beer? Beer { get; set; }
}

File: Frontend/BeerJournalFrontend/src/app/models/tasting-entry.ts
// TastingEntry — model (defines the shape of tasting entry data used in the app)

import { Beer } from './beer';

export interface TastingEntry {
  entryId: number;
  userId: number;

  // Foreign key — what the client SENDS when creating/updating
  beerId: number;

  // Tasting info
  rating: number | null;
  location: string | null;
  testingDate: string | null;   // ISO date string e.g. "2026-05-02"
  price: number | null;
  notes: string | null;


  // Map coordinates
  latitude: number | null;
  longitude: number | null;

  // Nested beer — populated by backend on GET (via JOIN)
  // Optional because on POST/PUT we only send beerId
  beer?: Beer;
}